core/src/main/java/oracle/weblogic/deploy/util/WLSDeployArchive.java

```diff
@@ -201,9 +228,11 @@ public void addModel(File model) throws WLSDeployArchiveIOException {
         final String METHOD = "addModel";
 
         LOGGER.entering(CLASS, METHOD, model);
-        validateExistingFile(model, "model", getArchiveFileName(), METHOD);
-        getZipFile().removeZipEntries(ARCHIVE_MODEL_TARGET_DIR);
-        addItemToZip(ARCHIVE_MODEL_TARGET_DIR, model);
+        if (hasArchive) {
```

Comment:
With a code block this small, it doesn't matter much.  But, code analyzers tend to report that the quick test and return are less "complex" for the reader.  Like
if (hasArchive) {
  return;
}

Reply:
I would agree with that except that all of these methods have enter / exit logging